camviewer_ui_impl.py

```diff
@@ -1297,6 +1307,44 @@ def onSizeUpdate(self):
         except Exception:
             pass
 
+    def user_set_max_image_rate(self, rate: int) -> None:
+        """
+        Call set_max_image_rate and record the value as the desired rate.
+
+        This desired rate is used so that after a timeout, we can
+        restore the GUI to the user's last desired rate when they
+        interact with the GUI again.
+        """
+        self.last_des_max_rate = rate
+        self.set_max_image_rate(rate)
+
+    def set_max_image_rate(self, rate: int) -> None:
+        """
+        Update the maximum allowed rate by restarting the appropriate timer.
+
+        Rate is expected to be a positive integer in Hz.
+
+        When this timer expires, wantImage is called.
```

Comment:
Is this true?  looks like only stop/setDisco are called

Reply:
`wantImage` is called when `imageTimer` times out, it's set up somewhere else.

Reply:
Ah ok I got mixed up here.  The QTimers aren't blocking, and `wantImage` is connected to `ImageTimer`.  stop/setDisco are called without waiting for this timer